ports/rp2/mpconfigport.h

```diff
@@ -218,6 +218,10 @@ extern const struct _mp_obj_type_t mod_network_nic_type_nina;
 #if MICROPY_PY_NETWORK_WIZNET5K
 extern const struct _mp_obj_type_t mod_network_nic_type_wiznet5k;
 #define MICROPY_HW_NIC_WIZNET5K             { MP_ROM_QSTR(MP_QSTR_WIZNET5K), MP_ROM_PTR(&mod_network_nic_type_wiznet5k) },
+// This Network interface requires the extended socket state for timeouts and non-blocking
```

Comment:
I think the extended state is only needed for UDP sendto functionality?

Reply:
I think so.

The additional state is for POSIX semantics using `connect` with a UDP socket. For `connect`, then remote host and port are stashed in the extended state and used for future calls to `send`. The `sendto` method continues to require a remote address for each call.

Do I need a better comment there?

Reply:
I think it needs a more accurate comment, because when I read this comment I was confused.

Actually, instead of repeating the comment in each port that uses this, it would be better to just enable the extended sockets here in the `mpconfigport.h` file without any comment, and then in `extmod/network_wiznet5k.c` have a check for this option and a comment, something like:
```c
// Requires extended socket state for send...
#if !MICROPY_PY_SOCKET_EXTENDED_STATE
#error WIZNET5K_PROVIDED_STACK requires MICROPY_PY_SOCKET_EXTENDED_STATE
#endif
```